app/controllers/smart_answers_controller.rb

```diff
@@ -11,6 +11,12 @@ class SmartAnswersController < ApplicationController
   rescue_from SmartAnswer::FlowRegistry::NotFound, with: :error_404
   rescue_from SmartAnswer::InvalidNode, with: :error_404
 
+  content_security_policy only: :visualise do |p|
+    # The script used to render the visualise tool requires eval execution
+    # unfortunately
+    p.script_src(*p.script_src, :unsafe_eval)
```

Comment:
It's good this is specific to the controller action. Can we make it specific to the file executed, as well? (It looks like `*p.script_src` would apply to any file.)

Reply:
It doesn't actually work like that, but I can see how you came to that impression.

It's somewhat unclear from the Rails DSL but what this does is append an additional rule to those already set for script. We need an additional rule added for script which is unsafe_eval and that isn't specified to a particular host or script just a general rule.

The eventual output would be like: `script-src 'self' *.publishing.service.gov.uk *.www.gov.uk www.gov.uk *.dev.gov.uk www.google-analytics.com ssl.google-analytics.com stats.g.doubleclick.net www.signin.service.gov.uk *.ytimg.com www.youtube.com www.youtube-nocookie.com hmrc-uk.digital.nuance.com 'unsafe-inline' 'unsafe-eval'`

Reply:
Ah, I see. Thanks for explaining. It looks like one [can have more than one CSP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#Multiple_content_security_policies), to accommodate multiple `script-src` tags, but only to make the policy stricter, which isn't compatible with the 'exception' scenario we have here. And [Rails doesn't support it](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/content_security_policy.rb#L166) anyway.

Reply:
Oh cool I didn't know you could do the multiple header approach.